lib/publishing/nuget/publish.sh

```diff
@@ -86,6 +86,13 @@ for NUGET_PACKAGE_PATH in $(find dotnet -name *.nupkg -not -iname *.symbols.nupk
             # The .snupkg will be published at the same time as the .nupkg if both are in the current folder (which is the case)
             $dotnet nuget push $NUGET_PACKAGE_NAME -k $NUGET_API_KEY -s $NUGET_SOURCE | tee ${log}
         fi
+
+        # dotnet nuget push writes a stream of output. errors are prefixed with error:
+        # Alternatively we could check for success string "Your package was pushed."
+        if tail -n 1 ${log} | grep -q "error"; then
```

Comment:
This script runs *without* `-o pipefail`, and the `$dotnet nuget push` command above is piped to `tee ${log}`, meaning if the `$dotnet nuget push` command fails, it's exist status will be lost (in favor of `tee`'s, which is likely to always succeed).

Instead of doing this, I would suggest using the `${PIPESTATUS[n]}` array to check the exist status of the `$dotnet nuget push` command.

Now, we don't enable `-o pipefail` in order to preserve idempotency, and as-is, you'd also get an error if attempting to publish a package that is already there (not good!). In such cases, `nuget push` offers the `-SkipDuplicate` option that'll allow you to simply skip that package and move on to the next one to push... I reckon this is all we need (and hence we ought to be able to enable `-o pipefail` at the top of the script instead of doing this dance...

Reply:
thanks @RomainMuller - that does sound a lot more robust. appreciate the guidance. 
updated - let me know if this is what you had in mind!